cmd/podman/images/scp.go

```diff
@@ -0,0 +1,332 @@
+
+	// TODO: add podman remote support
+
+	switch {
+	case scpOpts.FromRemote: // if we want to load FROM the remote
+		_, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.DestImageName, "", false)
+		if err != nil {
+			return err
+		}
+		if scpOpts.ToRemote { // we want to load remote -> remote
+			rep, err := execCommand(iden[1], uri[1], scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.Tag, true) // validate connection given and execute save or load on remote machine
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+			break
+		}
+		report, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+	case scpOpts.ToRemote: // remote host load
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.DestImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.Load.Input, "", scpOpts.ToRemote)
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	// TODO: Add podman remote support
+	default: // else native load
+		if scpOpts.Tag != "" {
+			return errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+		}
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.DestImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// execCommand takes ssh information and an image file path on the client, connects via ssh, executes
+// an scp command and then calls podman image load on the remote machine given the new scp filepath
+func execCommand(iden string, uri *urlP.URL, localFile string, image string, tag string, LoadOrSave bool) (string, error) {
+	cfg, err := connection.ValidateAndConfigure(uri, iden)
+	if err != nil {
+		return " ", err
+	}
+	dial, err := ssh.Dial("tcp", uri.Host, cfg) // dial the client
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to connect")
+	}
+	defer dial.Close()
+
+	// two sessions are necessary to create tmpDir on remote since only one run is allowed per session
+	sessions := []ssh.Session{}
+	for i := 0; i <= 1; i++ {
```

Comment:
Humans should not be manually counting whether to use `sessions[0]` or `sessions[1]`, just like we don’t keep all integer variables in an `intVariables[500]` array and manually allocate them to minimize overlap (We did! with assembler and to an extent FORTRAN and other languages of that era).

Hide this into an `execRemoteCommand` as discussed around https://github.com/containers/podman/pull/10828#discussion_r678672185 , and let this function worry only about which commands to run, not how.

(That will even be more efficient, because we don’t need the two sessions to be around in parallel.)